source/extensions/transport_sockets/http_11_proxy/connect.cc

```diff
@@ -29,12 +29,36 @@ bool UpstreamHttp11ConnectSocket::isValidConnectResponse(absl::string_view respo
 
 UpstreamHttp11ConnectSocket::UpstreamHttp11ConnectSocket(
     Network::TransportSocketPtr&& transport_socket,
-    Network::TransportSocketOptionsConstSharedPtr options)
-    : PassthroughSocket(std::move(transport_socket)), options_(options) {
-  if (options_ && options_->http11ProxyInfo() && transport_socket_->ssl()) {
+    Network::TransportSocketOptionsConstSharedPtr options,
+    std::shared_ptr<const Upstream::HostDescription> host, bool tls_exclusive,
+    absl::optional<std::string> proto_proxy_addr)
+    : PassthroughSocket(std::move(transport_socket)), options_(options),
+      tls_exclusive_(tls_exclusive) {
+
+  // If the TLS criteria can't be met, there will be no CONNECT proxying.
+  const bool tls_criteria_met = (tls_exclusive && transport_socket_->ssl()) || !tls_exclusive;
+  if (!tls_criteria_met) {
+    return;
+  }
+
+  if (options_ && options_->http11ProxyInfo()) {
+    // Historically, this transport socket could only be configured via filter state metadata and
+    // was required to wrap a TLS transport socket to add the CONNECT header. The original behavior
+    // for the metadata configuration (which we must maintain) would send a CONNECT header to the
+    // proxy address found in the filter state metadata on port 443. Therefore, the port is
+    // hardcoded in this case.
     header_buffer_.add(
         absl::StrCat("CONNECT ", options_->http11ProxyInfo()->hostname, ":443 HTTP/1.1\r\n\r\n"));
     need_to_strip_connect_response_ = true;
+    return;
+  }
+
+  // Since there's no filter state metadata providing a proxy address, we only want to add a CONNECT
+  // header if there was a proxy address configured in the transport socket's proto config.
+  if (proto_proxy_addr.has_value()) {
+    header_buffer_.add(
+        absl::StrCat("CONNECT ", host->address()->asStringView(), " HTTP/1.1\r\n\r\n"));
+    need_to_strip_connect_response_ = true;
```

Comment:
After checked the source code of http_11_proxy, I am confused now about how this PR works. The http11ProxyInfo() will directs the host to create connection to the proxy address.

But how your configured address works?

/wait-any